packages/utils/src/lib/git.integration.test.ts

```diff
@@ -16,3 +24,66 @@ describe('getLatestCommit', () => {
     );
   });
 });
+
+describe('branchHasChanges', () => {
+  it('should return true if some changes are given', async () => {
+    await makeStatusDirty();
+    await expect(branchHasChanges()).resolves.toBe(true);
+    await makeStatusClean();
+  });
+  it('should return false if no changes are given', async () => {
+    await expect(branchHasChanges()).resolves.toBe(false);
+  });
+});
+
+describe('guardAgainstDirtyRepo', () => {
+  it('should throw if history is dirty', async () => {
+    await makeStatusDirty();
+    await expect(guardAgainstDirtyRepo()).rejects.toThrow(
+      'Repository should be clean before we you can proceed',
+    );
+    await makeStatusClean();
+  });
+  it('should not throw if history is clean', async () => {
+    await expect(guardAgainstDirtyRepo()).resolves.toEqual(void 0);
+  });
+});
+
+describe('getCurrentBranchOrTag', () => {
+  it('should log current branch', async () => {
+    await expect(getCurrentBranchOrTag()).resolves.toEqual(expect.any(String));
```

Comment:
Is there a way of knowing/setting the branch? `any(String)` is a weak assertion.

Reply:
min length or regex maybe?

Reply:
Is it not possible to determine which exact branch it is if the repo is created for tests only?

Reply:
Yeah, that's the key factor, related to this thread - https://github.com/code-pushup/cli/pull/469#discussion_r1479469141

If it's the actual branch, then the tests shouldn't assume anything - I could have uncommitted changes, I could be in a detached state in which case there is no branch, I definitely don't want to a test to start switching my branch or deleting untracked files - that's why I don't want the test to depend on the real repo in any way.

If it is a temporary repo, then yes, the test can set any state it wants (without interfering with local development) and have way more specific assertions.